### PR TITLE
Ndmis reports wite to csv

### PR DESCRIPTION
**Pull request by rgforsyth**
## What does this pull request do?

Performs the write to csv part of the ndmis performance report. Then publishes the csv to s3.

## What is the intent behind these changes?

To create 3 seperate performance report csv's for referrals, appointments and complexity

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/BatchUtils.kt [hunk 1]
```diff
@@ -54,14 +88,6 @@ class HeaderWriter(private val header: String) : FlatFileHeaderCallback {
   }
 }
 
-class LoggingWriter<T> : ItemWriter<T> {
```
Comment (tomsmyers) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/BatchUtils.kt:57]: i think we can leave this in - it's useful for developing future jobs i think

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfiguration.kt [hunk 2]
```diff
@@ -57,45 +106,59 @@ class NdmisPerformanceReportJobConfiguration(
...
       .processor(processor)
-      .writer(LoggingWriter<List<AppointmentData>>())
+      .writer(writer)
       .build()
   }
+
+  private val pushToS3Step = stepBuilderFactory["pushToS3Step"]
+    .tasklet { _: StepContribution, _: ChunkContext ->
+      listOf(referralReportPath, complexityReportPath, appointmentsReportPath).forEach { path ->
+        s3Service.publishFileToS3(ndmisS3Bucket, path, "export/csv/reports/")
```
Comment (tomsmyers) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfiguration.kt:158]: the current path we are putting the csv file is `dfinterventions/dfi/csv/reports/`. do we need to put these in the same place? leigh mentioned we will have some overlap, so probably not. but we should ask where the ndmis team want them in the bucket.

**Review by tomsmyers — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfiguration.kt (hunk 2 above)
Comment (rgforsyth) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfiguration.kt:158]: the ticket says: 
Upload those files to S3 bucket (confirmed)
export/csv/reports/crs_performance_report-v2-referrals.csv

**Pull request merged**